code/modules/mob/living/simple_animal/worm.dm

```diff
@@ -44,154 +44,154 @@
 	var/atom/currentlyEating //what the worm is currently eating
 	var/eatingDuration = 0 //how long he's been eating it for
 
-	head
-		name = "space worm head"
-		icon_state = "spacewormhead"
-		icon_dead = "spaceworm_dead"
+/mob/living/simple_animal/space_worm/head
+	name = "space worm head"
+	icon_state = "spacewormhead"
+	icon_dead = "spaceworm_dead"
 
-		maxHealth = 20
-		health = 20
+	maxHealth = 20
+	health = 20
 
-		melee_damage_lower = 10
-		melee_damage_upper = 15
-		attacktext = "bitten"
+	melee_damage_lower = 10
+	melee_damage_upper = 15
+	attacktext = "bitten"
 
-		animate_movement = SLIDE_STEPS
+	animate_movement = SLIDE_STEPS
 
-		New(var/location, var/segments = 6)
-			..()
+/mob/living/simple_animal/space_worm/head/New(var/location, var/segments = 6)
+	. = ..()
 
-			var/mob/living/simple_animal/space_worm/current = src
+	var/mob/living/simple_animal/space_worm/current = src
 
-			for(var/i = 1 to segments)
-				var/mob/living/simple_animal/space_worm/newSegment = new /mob/living/simple_animal/space_worm(loc)
-				current.Attach(newSegment)
-				current = newSegment
+	for(var/i = 1 to segments)
+		var/mob/living/simple_animal/space_worm/newSegment = new /mob/living/simple_animal/space_worm(loc)
+		current.Attach(newSegment)
+		current = newSegment
 
-		update_icon()
-			if(stat == CONSCIOUS || stat == UNCONSCIOUS)
-				icon_state = "spacewormhead[previous?1:0]"
-				if(previous)
-					set_dir(get_dir(previous,src))
-			else
-				icon_state = "spacewormheaddead"
-
-	Life()
-		..()
-
-		if(next && !(next in view(src,1)))
-			Detach()
-
-		if(stat == DEAD) //dead chunks fall off and die immediately
-			if(previous)
-				previous.Detach()
-			if(next)
-				Detach(1)
-
-		if(prob(stomachProcessProbability))
-			ProcessStomach()
+/mob/living/simple_animal/space_worm/head/update_icon()
+	if(stat == CONSCIOUS || stat == UNCONSCIOUS)
```

Comment:
Ditto here

Reply:
@TorinoFermic that's bad code formatting but it's valid check. Right now stat can be `CONSCIOUS`, `UNCONSCIOUS` and `DEAD`. Actually that check means `if(stat != DEAD)`

Reply:
Isn't my code, it was already here.